src/vs/workbench/parts/debug/node/debugAdapter.ts

```diff
@@ -28,34 +29,74 @@ export class Adapter {
 		}
 	}
 
-	public get runtime(): string {
+	public getAdapterExecutable(verifyAgainstFS = true): TPromise<IAdapterExecutable> {
+
+		if (this.rawAdapter.adapterExecutableCommand) {
+			return this.commandService.executeCommand<IAdapterExecutable>(this.rawAdapter.adapterExecutableCommand).then(ad => {
+				return this.verifyAdapterDetails(ad, verifyAgainstFS);
+			});
+		}
+
+		const ad = <IAdapterExecutable>{
+			command: this.getProgram(),
+			args: this.getAttributeBasedOnPlatform('args')
+		};
+		const runtime = this.getRuntime();
+		if (runtime) {
+			const runtimeArgs = this.getAttributeBasedOnPlatform('runtimeArgs');
+			ad.args = (runtimeArgs || []).concat([ad.command]).concat(ad.args || []);
+			ad.command = runtime;
+		}
+		return this.verifyAdapterDetails(ad, verifyAgainstFS);
+	}
+
```

Comment:
It feels to me like `verifyAgainstFS` should not be passed to this method, but if that is `false` this method should simply not be called.